openchain/consensus/obcpbft/pbft-core.go

```diff
@@ -218,51 +209,7 @@ func newPbftCore(id uint64, config *viper.Viper, consumer innerStack, ledger con
 	// initialize state transfer
 	instance.hChkpts = make(map[uint64]uint64)
 
-	defaultPeerIDs := make([]*protos.PeerID, instance.replicaCount-1)
-	if instance.replicaCount > 1 {
-		// For some tests, only 1 replica will be present, and defaultPeerIDs makes no sense
-		for i := uint64(0); i < uint64(instance.replicaCount); i++ {
-			handle, err := getValidatorHandle(i)
-			if err != nil {
-				panic(fmt.Errorf("Cannot retrieve handle for peer which must exist : %s", err))
-			}
-			if i < instance.id {
-				logger.Debug("Replica %d assigning %v to index %d for replicaCount %d and id %d", instance.id, handle, i, instance.replicaCount, instance.id)
-				defaultPeerIDs[i] = handle
-			} else if i > instance.id {
-				logger.Debug("Replica %d assigning %v to index %d for replicaCount %d and id %d", instance.id, handle, i-1, instance.replicaCount, instance.id)
-				defaultPeerIDs[i-1] = handle
-			} else {
-				// This is our ID, do not add it to the list of default peers
-			}
-		}
-	} else {
-		logger.Debug("Replica %d not initializing defaultPeerIDs, as replicaCount is %d", instance.id, instance.replicaCount)
-	}
-
-	if myHandle, err := getValidatorHandle(instance.id); err != nil {
-		panic("Could not retrieve own handle")
-	} else {
-		instance.sts = statetransfer.NewStateTransferState(myHandle, config, ledger, defaultPeerIDs)
-	}
-
-	listener := struct{ statetransfer.ProtoListener }{}
-	listener.CompletedImpl = instance.stateTransferCompleted
-	instance.sts.RegisterListener(&listener)
-
-	// load genesis checkpoint
-	genesisBlock, err := instance.ledger.GetBlock(0)
-	if err != nil {
-		panic(fmt.Errorf("Cannot load genesis block: %s", err))
-	}
-	genesisHash, err := ledger.HashBlock(genesisBlock)
-	if err != nil {
-		panic(fmt.Errorf("Cannot hash genesis block: %s", err))
-	}
-	instance.chkpts[0] = &blockState{
-		blockNumber: 0,
-		blockHash:   base64.StdEncoding.EncodeToString(genesisHash),
-	}
+	instance.chkpts[0] = base64.StdEncoding.EncodeToString(startupID)
```

Comment:
Partially, assuming that all VPs (or at least 2f+1 honest ones) were stopped while idle, so that their state, and chain were the same, then this should allow us to handle the restart.

However, this is no different than the previous behavior, but PBFT takes its initial checkpoint value via the constructor, rather than interrogating the ledger directly.